java-frontend/src/main/java/org/sonar/java/regex/ast/DisjunctionTree.java

```diff
@@ -30,7 +30,7 @@
   private final List<JavaCharacter> orOperators;
 
   public DisjunctionTree(RegexSource source, IndexRange range, List<RegexTree> alternatives, List<JavaCharacter> orOperators) {
-    super(source, range);
+    super(source, range, FLAGS_HAVE_NO_EFFECT_ON_THIS_STATE);
```

Comment:
If we decide to remove `RegexParseResult.initialFlags`, `RegexTree`s should never use `FLAGS_HAVE_NO_EFFECT_ON_THIS_STATE`.

Reply:
I have set correct activeFlags on all AutomatonState and removed FLAGS_HAVE_NO_EFFECT_ON_THIS_STATE